lib/tilebag.rb

```diff
@@ -51,18 +42,23 @@ def initialize
 # turn @quantity into an array, shuffle, pop, and return x amount of letters
 
     def draw_tiles(num)
+      if num > 7
+        return []
+      else
       letters_array = []
         num.times do
         letters_array.push(@quantity.keys.shuffle.pop)
         end
-
         letters_array.each do |letter|
-          quantity.find
-
+          @quantity[letter] -= 1
         end
+        return letters_array
+      end
+    end
 
-      return letters_array
-
+    def tiles_remaining
+      sum = @quantity.values.inject(0) {|sum, value| sum + value}
+      sum
```

Comment:
The line above would be sufficient in returning the `sum` variable value. I would recommend that if you want/need to do an explicit return, use the `return` keyword to make it a bit more clear